nixos/modules/services/networking/wireguard.nix

```diff
@@ -201,45 +225,14 @@ let
         };
 
         script = ''
-          modprobe wireguard
-
-          ${values.preSetup}
-
-          ip link add dev ${name} type wireguard
-
-          ${concatMapStringsSep "\n" (ip:
-            "ip address add ${ip} dev ${name}"
-          ) values.ips}
-
-          wg set ${name} private-key ${privKey} ${
-            optionalString (values.listenPort != null) " listen-port ${toString values.listenPort}"}
-
+          wg set ${name} private-key ${values.privateKeyFile}
```

Comment:
This should be wrapped in `optionalString (values.privateKeyFile != null)` to handle the use of `privateKey`

Reply:
sure.